optimize/backend/src/main/java/io/camunda/optimize/TomcatConfig.java

```diff
@@ -0,0 +1,273 @@
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRegistration;
+import java.net.URL;
+import java.util.Optional;
+import org.apache.catalina.connector.Connector;
+import org.apache.coyote.http2.Http2Protocol;
+import org.apache.tomcat.util.net.SSLHostConfig;
+import org.apache.tomcat.util.net.SSLHostConfigCertificate;
+import org.apache.tomcat.util.net.SSLHostConfigCertificate.Type;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.server.WebServerFactoryCustomizer;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.boot.web.servlet.ServletContextInitializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+@Configuration
+public class TomcatConfig {
+
+  public static final String EXTERNAL_SUB_PATH = "/external";
+
+  private static final Logger LOG = org.slf4j.LoggerFactory.getLogger(TomcatConfig.class);
+
+  private static final String[] COMPRESSED_MIME_TYPES = {
+    "application/json", "text/html", "application/x-font-ttf", "image/svg+xml"
+  };
+
+  private static final String LOGIN_ENDPOINT = "/login";
+  private static final String METRICS_ENDPOINT = "/metrics";
+  private static final String URL_BASE = "/#";
+
+  public static final String ALLOWED_URL_EXTENSION =
+      String.join(
+          "|",
+          new String[] {
+            URL_BASE,
+            LOGIN_ENDPOINT,
+            METRICS_ENDPOINT,
+            CCSaasAuth0WebSecurityConfig.OAUTH_AUTH_ENDPOINT,
+            CCSaasAuth0WebSecurityConfig.OAUTH_REDIRECT_ENDPOINT,
+            CCSaasAuth0WebSecurityConfig.AUTH0_JWKS_ENDPOINT,
+            CCSaasAuth0WebSecurityConfig.AUTH0_AUTH_ENDPOINT,
+            CCSaasAuth0WebSecurityConfig.AUTH0_TOKEN_ENDPOINT,
+            CCSaasAuth0WebSecurityConfig.AUTH0_USERINFO_ENDPOINT,
+            HealthRestService.READYZ_PATH,
+            LocalizationRestService.LOCALIZATION_PATH,
+            TomcatConfig.EXTERNAL_SUB_PATH,
+            OptimizeResourceConstants.REST_API_PATH,
+            OptimizeResourceConstants.STATIC_RESOURCE_PATH,
+            OptimizeResourceConstants.ACTUATOR_ENDPOINT,
+            PanelNotificationConstants.SEND_NOTIFICATION_TO_ALL_ORG_USERS_ENDPOINT,
+            RestConstants.BACKUP_ENDPOINT,
+            UIConfigurationRestService.UI_CONFIGURATION_PATH
+          });
+
+  private static final String HTTP11_NIO_PROTOCOL = "org.apache.coyote.http11.Http11Nio2Protocol";
+
+  @Autowired private ConfigurationService configurationService;
+
+  @Autowired private Environment environment;
+
+  @Bean
+  WebServerFactoryCustomizer<TomcatServletWebServerFactory> tomcatFactoryCustomizer() {
+    LOG.debug("Setting up connectors...");
```

Comment:
```suggestion
    LOG.debug("Setting up Server connectors...");
```
Nitpicky, but feels appropriate given that connectors within Camunda means a very different thing